sdk/core/Azure.Core.Experimental/src/Amqp/AmqpMessageHeader.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Amqp
+{
+    /// <summary>
+    /// Represents an AMQP message header.
+    /// http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-header
+    /// </summary>
+    public class AmqpMessageHeader
+    {
+        /// <summary>
+        /// Initializes a new <see cref="AmqpMessageHeader"/> instance.
+        /// </summary>
+        public AmqpMessageHeader() { }
+
+        internal AmqpMessageHeader(AmqpMessageHeader header)
```

Comment:
nit:  Comments on non-public members are appreciated to help contributors (internal and external) further their understandings.   :smile: